File: robocode2-bot-api-java/src/main/java/net/robocode2/events/BulletHitWallEvent.java
package net.robocode2.events;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import lombok.Value;
import net.robocode2.BulletState;

/** Event occurring when a bullet has hit a wall */
@Value
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public class BulletHitWallEvent extends Event {
  /** Bullet that has hit a wall */
  BulletState bullet;

  @Builder
  private BulletHitWallEvent(int turnNumber, BulletState bullet) {
    this.turnNumber = turnNumber;
    this.bullet = bullet;
  }
}

File: bot-api/python/src/robocode_tank_royale/bot_api/events/bullet_hit_wall_event.py
from dataclasses import dataclass

from ..bullet_state import BulletState
from .bot_event import BotEvent


@dataclass(frozen=True, repr=True)
class BulletHitWallEvent(BotEvent):
    """
    Represents an event that occurs when a bullet hits a wall during a game.

    This event provides information about which bullet hit the wall and at
    what turn the event occurred.

    Attributes:
        bullet (BulletState): The bullet that hit the wall.

    """

    bullet: BulletState
